activesupport/lib/active_support/core_ext/string/output_safety.rb

```diff
@@ -136,15 +136,39 @@ def encode_with(coder)
 
     UNSAFE_STRING_METHODS.each do |unsafe_method|
       if unsafe_method.respond_to?(unsafe_method)
+        arity = unsafe_method.method(unsafe_method).arity
+        params =
+          case arity
+          when 0
+            "&block"
+          when 1
+            "arg, &block"
+          when (2..)
+            1.upto(arity).each_with_object(+"") { |i, s| s << "arg#{i}, " } << "&block"
+          else
+            "*args, &block"
+          end
```

Comment:
I think we could simplify this to

```suggestion
        param_names = arity < 0 ? ["*args"] : 1.upto(arity).map { |i| "arg#{i}" }
        params = param_names.append("&block").join(", ")
```

Reply:
Naming the argument `arg1` where having only one argument seems a little bit odd to me. Sure it's just about internal `eval`ing code, but the users may refer to this by inspecting the method object, like so:
```
$ rails r "p ''.html_safe.method(:delete_suffix)"
#<Method: ActiveSupport::SafeBuffer#delete_suffix(arg, &block) .../activesupport/lib/active_support/core_ext/string/output_safety.rb:164>
```

If we don't care about that, we could code golf as you suggest.